src/X11.cc

```diff
@@ -813,21 +814,48 @@ X11::getScreenChangeNotification(XEvent *ev, ScreenChangeNotification &scn)
 //! @return The nearest head.  Head numbers are indexed from 0.
 uint
 X11::getNearestHead(int x, int y)
+{
+	uint nearest_head;
+	if (!getNearestHead(x, y, nearest_head)) {
+		nearest_head = 0;
+	}
+	return nearest_head;
+}
+
+//! @brief Searches for the head closest to the coordinates x,y.
+//!
+//! *dx and *dy contain direction values -1 (west or north), 0 and 1
+//! (east or south). If not null, only the nearest window in that
+//! direction is selected.
+//!
+//! @return The nearest head.  Head numbers are indexed from 0.
+bool
+X11::getNearestHead(int x, int y, uint& nearest_head, int* dx, int* dy)
```

Comment:
s/int/enum DirectionType/g to make it a bit more clear.

Also, could we change from a pointer and instead add another value to the type as the skip value?